toolchain/sem_ir/value_stores.h

```diff
@@ -146,17 +146,35 @@ class ConstantValueStore : public Yaml::Printable<ConstantValueStore> {
   llvm::SmallVector<ConstantId, 0> values_;
 };
 
-// Provides storage for instructions representing global constants.
+// Provides storage for instructions representing deduplicated global constants.
 class ConstantStore {
  public:
-  // Add a constant instruction.
-  auto Add(InstId inst_id) -> void { values_.push_back(inst_id); }
+  explicit ConstantStore(SemIR::File* sem_ir, llvm::BumpPtrAllocator& allocator)
+      : allocator_(&allocator), constants_(sem_ir) {}
```

Comment:
```suggestion
  explicit ConstantStore(SemIR::File& sem_ir, llvm::BumpPtrAllocator& allocator)
      : allocator_(&allocator), constants_(&sem_ir) {}
```

It seems odd to me that you take in one as pointer, and the other as reference. Maybe both as reference?

Reply:
Fixed.